sql/src/main/java/io/crate/expression/reference/sys/check/cluster/TablesNeedUpgradeSysCheck.java

```diff
@@ -53,13 +55,12 @@
                                        "WHERE min_lucene_version not like '" + Version.LATEST.major + ".%.%' " +
                                        "GROUP BY 1, 2 " +
                                        "ORDER BY 1";
-    private static final int LIMIT = 50_000;
-    private static final String PREP_STMT_NAME = "tables_need_upgrade_syscheck";
+    private static final Statement PARSED_STMT = SqlParser.createStatement(STMT);
 
     private final Logger logger;
     private final Provider<SQLOperations> sqlOperationsProvider;
-    private SQLOperations.SQLDirectExecutor sqlDirectExecutor;
     private volatile Collection<String> tablesNeedUpgrade;
+    private Session session;
```

Comment:
Can we initialize in ctor and avoid: https://github.com/crate/crate/pull/6902/files#diff-5e7b03a1e21d2edcb33fbd1b2f9e8155R97

Reply:
Unfortunately no - it causes a cyclic dependency injection error

Reply:
ah ok, Thx!